aws/src/main/java/org/apache/iceberg/aws/s3/S3InputStream.java

```diff
@@ -178,18 +183,23 @@ private void positionStream() throws IOException {
   }
 
   private void openStream() throws IOException {
-    GetObjectRequest.Builder requestBuilder =
-        GetObjectRequest.builder()
-            .bucket(location.bucket())
-            .key(location.key())
-            .range(String.format("bytes=%s-", pos));
-
-    S3RequestUtil.configureEncryption(s3FileIOProperties, requestBuilder);
-
     closeStream();
 
     try {
-      stream = s3.getObject(requestBuilder.build(), ResponseTransformer.toInputStream());
+      stream =
+          RetryableInputStream.builderFor(
+                  rangeStart -> {
+                    GetObjectRequest.Builder requestBuilder =
+                        GetObjectRequest.builder()
+                            .bucket(location.bucket())
+                            .key(location.key())
+                            .range(String.format("bytes=%s-", rangeStart));
+                    S3RequestUtil.configureEncryption(s3FileIOProperties, requestBuilder);
+                    return s3.getObject(
+                        requestBuilder.build(), ResponseTransformer.toInputStream());
+                  },
+                  () -> pos)
```

Comment:
Discussed with @danielcweeks, we can just simplify this to remove the extra position supplier parameter to the builderFor API and just pass in the pos as part of the stream supplier